packages/firebase/index.d.ts

```diff
@@ -5484,6 +5484,87 @@ declare namespace firebase.storage {
      *     including if the object did not exist.
      */
     updateMetadata(metadata: firebase.storage.SettableMetadata): Promise<any>;
+    /**
+     * List all items (files) and prefixes (folders) under this storage reference.
+     *
+     * This is a helper method for calling list() repeatedly until there are
+     * no more results. The default pagination size is 1000.
+     *
+     * Note: The results may not be consistent if objects are changed while this
+     * operation is running.
+     *
+     * Warning: listAll may potentially consume too many resources if there are
+     * too many results.
+     *
+     * @return A Promise that resolves with all the items and prefixes under
+     *      the current storage reference. `prefixes` contains references to
+     *      sub-directories and `items` contains references to objects in this
+     *      folder. `nextPageToken` is never returned.
+     */
+    listAll(): Promise<ListResult>;
+    /**
+     * List items (files) and prefixes (folders) under this storage reference.
+     *
+     * List API is only available for Firebase Rules Version 2.
+     *
+     * GCS is a key-blob store. Firebase Storage imposes the semantic of '/'
+     * delimited folder structure.
+     * Refer to GCS's List API if you want to learn more.
+     *
+     * To adhere to Firebase Rules's Semantics, Firebase Storage does not
+     * support objects whose paths end with "/" or contain two consecutive
+     * "/"s. Firebase Storage List API will filter these unsupported objects.
+     * list() may fail if there are too many unsupported objects in the bucket.
+     *
+     * @param options.maxResults If set, limits the total number of `prefixes`
+     *      and `items` to return. The default and maximum maxResults is 1000.
+     *      Use the nextPageToken to retrieve more objects.
+     * @param options.pageToken The `nextPageToken` from a previous call to
```

Comment:
This style of describing the parameter only works if you define the type directly in this method, like the following:
```javascript
    list(options?: {maxResults: number | null, pageToken: string | null}
```
If an interface is used, the comments should coexist with the interface definition.

Since you have defined an interface, can you move the comments to the interface?

Reply:
Done. Updated the doc.